Cleanup zha.markdown hardware list and add config for ti_cc (#12233)
home-assistant/home-assistant.github.io: source/_integrations/zha.markdown

  - '@adminiuga'
---

> [Zigbee Home Automation (ZHA)](https://www.home-assistant.io/integrations/zha/)

Reviewer: It is now linking to itself?

Author: I don't think that the title of the integration should itself link to zigbeealliance.org since the implementation of not an official Zigbee Alliance integration. Maybe otherwise just remove that link completely?

Reviewer: Yeah, that sounds better. Linking to the page you are already at, is a bit useless :)

> integration nativly adds built-in hub support to Home Assistant for connecting many off-the-shelf [Zigbee](https://zigbeealliance.org) devices made for home automation, allowing Assistant itself to create its own Zigbee mesh network. Hardware-wise you just need one of the many compatible Zigbee coordinator radio adapters/modules directly connected to your computer runnning Home Assistant via USB or GPIO/UART.

Reviewer: ```suggestion
integration natively adds built-in support to Home Assistant for connecting many off-the-shelf [Zigbee](https://zigbeealliance.org) devices made for home automation, allowing Home Assistant to create its own Zigbee mesh network. Hardware-wise you just need one of the many compatible Zigbee coordinator radio adapters/modules directly connected to your computer running Home Assistant via USB or GPIO/UART.
```

- Sensor
- Switch

## Compatible hardware

> ZHA integration support using one of the multiple available Zigbee coordinator radio modules that is compatible with [zigpy](https://github.com/zigpy/zigpy). zigpy is an open source Python library implementing a hardware independent Zigbee stack that uses a modular design with separate Python libraries for Zigbee coordinator radio modules from different manufacturers, with each library capable of interfacing with USB and GPIO radio hardware adapters/modules over different UART serial communication protocols. Support for radio libraries in this ZHA integration include; [zigpy-deconz](https://github.com/zigpy/zigpy-deconz) for Dresden-Elektronik deCONZ based radios, [bellows](https://github.com/zigpy/bellows) for Silicon Labs EmberZNet/EZSP based radios, and [zigpy-xbee](https://github.com/zigpy/zigpy-xbee) for XBee based Zigbee radios. There are currently also experimental support for a couple additional radio libraries; [zigpy-cc](https://github.com/sanyatuning/zigpy-cc) for Texas Instruments Z-Stack based radios, and [zigpy-zigate](https://github.com/doudz/zigpy-zigate) for ZiGate based radios.

Reviewer: ```suggestion
The ZHA integration support using one of the multiple available Zigbee coordinator radio modules that are compatible with [zigpy](https://github.com/zigpy/zigpy). zigpy is an open source Python library implementing a hardware-independent Zigbee stack that uses a modular design with separate Python libraries for Zigbee coordinator radio modules from different manufacturers, with each library capable of interfacing with USB and GPIO radio hardware adapters/modules over different UART serial communication protocols. Support for radio libraries in this ZHA integration include; [zigpy-deconz](https://github.com/zigpy/zigpy-deconz) for Dresden-Elektronik deCONZ based radios, [bellows](https://github.com/zigpy/bellows) for Silicon Labs EmberZNet/EZSP based radios, and [zigpy-xbee](https://github.com/zigpy/zigpy-xbee) for XBee based Zigbee radios. There are currently also experimental support for a couple additional radio libraries; [zigpy-cc](https://github.com/sanyatuning/zigpy-cc) for Texas Instruments Z-Stack based radios, and [zigpy-zigate](https://github.com/doudz/zigpy-zigate) for ZiGate based radios.
```

> ZHA integration support using one of the multiple available Zigbee coordinator radio modules that is compatible with [zigpy](https://github.com/zigpy/zigpy). zigpy is an open source Python library implementing a hardware independent Zigbee stack that uses a modular design with separate Python libraries for Zigbee coordinator radio modules from different manufacturers, with each library capable of interfacing with USB and GPIO radio hardware adapters/modules over different UART serial communication protocols. Support for radio libraries in this ZHA integration include; [zigpy-deconz](https://github.com/zigpy/zigpy-deconz) for Dresden-Elektronik deCONZ based radios, [bellows](https://github.com/zigpy/bellows) for Silicon Labs EmberZNet/EZSP based radios, and [zigpy-xbee](https://github.com/zigpy/zigpy-xbee) for XBee based Zigbee radios. There are currently also experimental support for a couple additional radio libraries; [zigpy-cc](https://github.com/sanyatuning/zigpy-cc) for Texas Instruments Z-Stack based radios, and [zigpy-zigate](https://github.com/doudz/zigpy-zigate) for ZiGate based radios.

Reviewer: Honestly, this paragraph is too big IMHO and contains a lot of irrelevant information for the normal user.

The libraries and packages used, are not relevant for using the integration and might confuse a new user.

Author: Maybe, but just will notice that I removed the same links from the hardware list. So you could see it as if the information was just moved, summarized and clarified, so still fewer words all in all. Also, for troubleshooting, for specific Zigbee radio hardware issues the users would really need to submit issues and bug-reports upstream to each radio library repository.

Reviewer: For a generic user, they don't know. We have to take into account that our user base will consist of more people without a technical background nowadays.

I think it is fair to expect issue reports on Home Assistant if a user has problems using the integration. We can't expect them to dive into the codebase or to figure out which dependency is effected.

Author: Second opinions? My vote goes to keep that paragraph explain how the integration works. Again please note that I also removed the radio library reference from the hardware list so without this the information is not there. With this information being there users do not need to dive into the codebase as dependencies are listed.

Reviewer: A user of Home Assistant should not have to care about which library is used. That is the whole point I'm trying to make.

Added @klaasnicolaas as a second reviewer from the docs team for a second opinion.

Author: Please note I also wrote that sentence in order to try to explain to both end-users and potential developers wanting to help that support for even more Zigbee radio hardware from different manufacturers could be added to this integration. Meaning that there is no need to ask/request or make a new integration if you want support for new Zigbee radio hardware from different manufacturers, instead users can ask/request that support for additional Zigbee radio hardware can be added to this same integration even if that hardware comes from different manufacturers.

Author: OK I now removed half the paragraph including "Support for radio libraries in this ZHA integration include" and onwards but I then restored radio library links in hardware list.

Reviewer: Unresolving the conversation. Awaiting second opinion by @klaasnicolaas, as requested by yourself.

### Known working Zigbee radio modules

> - **Dresden Elektronik deCONZ based Zigbee radios**

Reviewer: "dresden elektronik" is written in lower case. Please respect the writing style of the brand.
```suggestion
- **dresden elektronik deCONZ based Zigbee radios**
```

If you are use ZiGate, you have to use some special usb_path configuration:

- ZiGate USB-TTL or USB-DIN: `/dev/ttyUSB0` or `auto` to auto discover the zigate
- PiZigate: `pizigate:/dev/serial0`
- ZiGate Pack WiFi: `socket://[IP]:[PORT]` for example `socket://192.168.1.10:9999`

{% configuration %}
radio_type:

>   description: One of `ezsp`, `xbee`, `deconz`, `zigate`, or `ti_cc`.

Reviewer: Not sure of the reasoning behind this change?
It was nicely sorted alphabetically, now it seems random?

Reset your Zigbee devices according to the device instructions provided by the manufacturer (e.g.,  turn on/off lights up to 10 times, switches usually have a reset button/pin).

## ZHA exception and deviation handling

Zigbee devices that deviate from or do not fully conform to the standard specifications set by the [Zigbee Alliance](https://zigbeealliance.org) may require the development of custom [ZHA Device Handlers](https://github.com/dmulcahey/zha-device-handlers) (ZHA custom quirks handler implementation) to for all their functions to work properly with the ZHA integration in Home Assistant. These ZHA Device Handlers for Home Assistant can thus be used to parse custom messages to and from Zigbee devices.

> The custom quirks implementations for zigpy implemented as ZHA Device Handlers for Home Assistant are a similar concept to that of [Hub-connected Device Handlers for the SmartThings Classics platform](https://docs.smartthings.com/en/latest/device-type-developers-guide/) as well as that of [Zigbee-Shepherd Converters as used by Zigbee2mqtt](https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html), meaning they are each virtual representations of a physical device that expose additional functionality that is not provided out-of-the-box by the existing integration between these platforms.

Reviewer: I think this paragraph can be removed as a whole.

Author: If you mean the paragraph about "ZHA exception and deviation handling" then I disagree, creating custom quirks (ZHA Device Handlers) for zhaquirks is something advanced Home Assistant end-users can do themselves in order to add support for devices that do not follow the standards. This is also something that is Home Assistant specific and not really genric to the zigpy library as I'm sure dmulcahey can attest to . Please see: https://github.com/dmulcahey/zha-device-handlers

Reviewer: An average user will have no clue on what this means even...

Author: Maybe but it is still specific to this ZHA integration in Home Assistant. Perhaps it should just be clarified that "advanced users" can add support for not working devices via ZHA Device Handlers? Please also note that this whole paragraph was not written now for this PR as I only moved the paragraph from the beginning to the end of the page, so perhaps that discussion belongs in a separate PR regardless.

Author: Accept to resolve this as a moot point for PR as that paragraph was just moved further down?

Author: @dmulcahey & @Adminiuga What are your opinion on keeping info on ZHA Device Handlers?

Reviewer: You are trying to raise a discussion in a resolved comment. Please unresolve it again in that case. We get the notification, but your comment isn't visible if the conversation is resolved. I've now unresolved it.

    restart: always
    network_mode: host
```

### Add Philips Hue bulbs that have previously been added to another bridge

> Philips Hue bulbs that have previously been added to another bridge won't show up during search. You have to restore your bulbs back to factory settings first. To achieve this, you basically have the following options.

Reviewer: ```suggestion
Philips Hue bulbs that have previously been added to another bridge won't show up during a search. You have to restore your bulbs back to factory settings first. To achieve this, you basically have the following options.
```

#### Philips Hue Dimmer Switch

Using a Philips Hue Dimmer Switch is probably the easiest way to factory-reset your bulbs. For this to work, the remote doesn't have to be paired with your previous bridge.

> 1. Turn on your Hue bulb you want to reset

Reviewer: ```suggestion
1. Turn on your Hue bulb you want to reset.
```

> 2. Hold the Dimmer Switch near your bulb (< 10 cm)

Reviewer: ```suggestion
2. Hold the Dimmer Switch near your bulb (< 10 cm).
```

> 3. Press and hold the (I)/(ON) and (O)/(OFF) buttons of the Dimmer Switch for about 10 seconds until your bulb starts to blink

Reviewer: ```suggestion
3. Press and hold the (I)/(ON) and (O)/(OFF) buttons of the Dimmer Switch for about 10 seconds until your bulb starts to blink.
```

4. Your bulb should stop blinking and eventually turning on again. At the same time, a green light on the top left of your remote indicates that your bulb has been successfully reset to factory settings.

#### hue-thief

> Follow the instructions on [https://github.com/vanviegen/hue-thief/](https://github.com/vanviegen/hue-thief/) (EZSP-based Zigbee USB stick required)

Reviewer: ```suggestion
Follow the instructions on [https://github.com/vanviegen/hue-thief/](https://github.com/vanviegen/hue-thief/) (EZSP-based Zigbee USB stick required).
```
